src/main/java/core/basesyntax/dao/impl/AbstractDao.java

```diff
@@ -1,11 +1,55 @@
 package core.basesyntax.dao.impl;
 
+import core.basesyntax.exceptions.DataProcessingException;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
 
-public abstract class AbstractDao {
+public abstract class AbstractDao<T> {
     protected final SessionFactory factory;
 
     protected AbstractDao(SessionFactory sessionFactory) {
         this.factory = sessionFactory;
     }
+
+    public T create(T entity) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            transaction.commit();
+            return entity;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert Content entity", e);
```

Comment:
Looks really cool to use Generics, but ... in all classes whenever you had error - your message will be about "Content". How can you understand whether it is smth wrong with User, Message, Comment?